frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/status-card/status-card.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import { Gallery, GalleryItem } from '@patternfly/react-core';
+import { ALERTS_KEY } from '@console/internal/actions/dashboards';
+import AlertsBody from '@console/shared/src/components/dashboard/status-card/AlertsBody';
+import AlertItem from '@console/shared/src/components/dashboard/status-card/AlertItem';
+import { alertURL, PrometheusRulesResponse } from '@console/internal/components/monitoring';
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import HealthBody from '@console/shared/src/components/dashboard/status-card/HealthBody';
+import HealthItem from '@console/shared/src/components/dashboard/status-card/HealthItem';
+import { getAlerts } from '@console/shared/src/components/dashboard/health-card/utils';
+import { PrometheusResponse } from '@console/internal/components/graphs';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import {
+  DATA_RESILIENCY_QUERY,
+  STORAGE_HEALTH_QUERIES,
+  StorageDashboardQuery,
+} from '../../../../constants/queries';
+import { filterCephAlerts } from '../../../../selectors';
+import { getCephHealthState, getDataResiliencyState } from './utils';
+
+const CephAlerts = withDashboardResources(({ watchAlerts, stopWatchAlerts, alertsResults }) => {
+  React.useEffect(() => {
+    watchAlerts();
+    return () => {
+      stopWatchAlerts();
+    };
+  }, [watchAlerts, stopWatchAlerts]);
+
+  const alertsResponse = alertsResults.getIn([ALERTS_KEY, 'data']) as PrometheusRulesResponse;
+  const alertsResponseError = alertsResults.getIn([ALERTS_KEY, 'loadError']);
+  const alerts = filterCephAlerts(getAlerts(alertsResponse));
+
+  return (
+    <AlertsBody
+      isLoading={!alertsResponse}
+      error={alertsResponseError}
+      emptyMessage="No persistent storage alerts"
+    >
+      {alerts.length &&
```

Comment:
Why do we need to check for `alerts` array emptiness?

We could simply do:

```tsx
{alerts.map((alert) => <AlertItem key={alertURL(alert, alert.rule.id)} alert={alert} />)}
```

Reply:
The `ActivityBody` component shows "no persistent storage alerts" only when it gets a `falsy` value .
The map function will return [], which will be `truthy`.
That's why adding a condition on length.[
Reference ](https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/dashboard/status-card/AlertsBody.tsx#L25)

Reply:
Thanks, it's clear to me now.